src/Button/Button.component.tsx

```diff
@@ -3,113 +3,602 @@ import * as React from 'react';
 // VENDOR
 import classNames from 'classnames';
 import styled, { css, FlattenSimpleInterpolation } from 'styled-components';
-import { darken } from 'polished';
+import { transparentize } from 'polished';
 // THEME
 import { colors, fonts, sizes } from '../theme';
+import { Colors } from '../Typography/Typography.component';
+
+type Variant = 'primary' | 'outline' | 'minimal';
+
+type ButtonSize = 'xlarge' | 'large' | 'small' | 'xsmall';
+
+interface Theme {
```

Comment:
I refer to this in typography as a hue as it's not a color, but a treatment of a color. How does one get a dark, savvy cyan? Or is that not a permutation we support?